apm-agent-core/src/main/java/co/elastic/apm/agent/context/LifecycleListener.java

```diff
@@ -47,6 +47,34 @@
      */
     void start(ElasticApmTracer tracer);
 
+    /**
+     * Callback for when {@link ElasticApmTracer#pause()} has been called.
+     * <p>
+     * Typically, this method is used to reduce overhead on the application to a minimum. This can be done by cleaning
+     * up resources like object pools, as well as by avoiding tracing-related overhead.
+     * </p>
+     * <p>
+     * Exceptions thrown from this method are caught and handled so that they don't prevent further cleanup actions.
```

Comment:
maybe replace "are caught and handled" maybe "should be caught and handled", because we document how users should use this method.

Reply:
> because we document how users should use this method.

Not really... It's a listener interface so we document how users should implement this method. The tracer is using it in a single place. This is basically encourage implementors to throw informative Exceptions.